azurerm/resource_arm_key_vault_secret.go

```diff
@@ -0,0 +1,240 @@
+package azurerm
+
+import (
+	"fmt"
+	"log"
+	"net/url"
+	"regexp"
+	"strings"
+
+	"github.com/Azure/azure-sdk-for-go/dataplane/keyvault"
+	"github.com/hashicorp/terraform/helper/schema"
+	"github.com/jen20/riviera/azure"
+)
+
+func resourceArmKeyVaultSecret() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceArmKeyVaultSecretCreate,
+		Read:   resourceArmKeyVaultSecretRead,
+		Update: resourceArmKeyVaultSecretUpdate,
+		Delete: resourceArmKeyVaultSecretDelete,
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: validateKeyVaultSecretName,
+			},
+
+			"vault_uri": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+
+			"value": {
+				Type:      schema.TypeString,
+				Required:  true,
+				Sensitive: true,
+			},
+
+			"content_type": {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+
+			"version": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+
+			"tags": tagsSchema(),
+		},
+	}
+}
+
+func resourceArmKeyVaultSecretCreate(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*ArmClient).keyVaultManagementClient
+
+	log.Print("[INFO] preparing arguments for AzureRM KeyVault Secret creation.")
+
+	name := d.Get("name").(string)
+	keyVaultBaseUrl := d.Get("vault_uri").(string)
+	value := d.Get("value").(string)
+	contentType := d.Get("content_type").(string)
+	tags := d.Get("tags").(map[string]interface{})
```

Comment:
If `tags` is optional, this cast will panic on a nil `"tags"` block: 
```
tags { }
```
With the complex types here, easier to try the cast to verify against potential panics.